paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/AddPaymentMethodInteractor.kt

```diff
@@ -0,0 +1,234 @@
+                    arguments = newFormArguments,
+                    formElements = newFormElements,
+                    linkInlineSignupMode = newLinkInlineSignupMode,
+                    usBankAccountFormArguments = newUsBankAccountFormArguments
+                )
+            }
+        }
+
+        coroutineScope.launch {
+            selection.collect {
+                _state.value = _state.value.copy(
+                    paymentSelection = it
+                )
+            }
+        }
+
+        coroutineScope.launch {
+            linkSignupMode.collect {
+                _state.value = _state.value.copy(
+                    linkSignupMode = it,
+                    linkInlineSignupMode = it.takeIf {
+                        shouldHaveLinkInlineSignup(_state.value.selectedPaymentMethodCode)
+                    }
+                )
+            }
+        }
+
+        coroutineScope.launch {
+            processing.collect {
+                _state.value = _state.value.copy(
+                    processing = it
+                )
+            }
+        }
+    }
+
+    override fun handleViewAction(viewAction: AddPaymentMethodInteractor.ViewAction) {
+        when (viewAction) {
+            is AddPaymentMethodInteractor.ViewAction.OnLinkSignUpStateUpdated -> onLinkSignUpStateUpdated(
+                viewAction.state
+            )
+            is AddPaymentMethodInteractor.ViewAction.ReportFieldInteraction -> reportFieldInteraction(
+                viewAction.code
+            )
+            is AddPaymentMethodInteractor.ViewAction.OnFormFieldValuesChanged -> onFormFieldValuesChanged(
+                viewAction.formValues,
+                viewAction.selectedPaymentMethodCode
+            )
+            is AddPaymentMethodInteractor.ViewAction.OnPaymentMethodSelected -> {
+                if (selectedPaymentMethodCode.value != viewAction.code) {
+                    _selectedPaymentMethodCode.value = viewAction.code
+                    reportPaymentMethodTypeSelected(viewAction.code)
+                }
+            }
+        }
+    }
+
+    override fun close() {
+        coroutineScope.cancel()
+    }
+
+    private fun shouldHaveLinkInlineSignup(selectedPaymentMethodCode: PaymentMethodCode): Boolean {
+        return selectedPaymentMethodCode == PaymentMethod.Type.Card.code
+    }
+}
+
+internal class UnsupportedAddPaymentMethodInteractor(private val errorReporter: ErrorReporter) :
+    AddPaymentMethodInteractor {
+
+    private val errorFieldFunctionAccessed = "function_accessed"
+
+    override val state: StateFlow<AddPaymentMethodInteractor.State>
+        get() {
+            errorReporter.report(
+                ErrorReporter.UnexpectedErrorEvent.UNSUPPORTED_ADD_PAYMENT_METHOD_INTERACTOR_USED,
+                additionalNonPiiParams = mapOf(errorFieldFunctionAccessed to "state")
+            )
+            throw UnsupportedOperationException("Attempting to use UnsupportedAddPaymentMethodInteractor")
```

Comment:
The error reporter stuff is not doing anything if we throw right after. It'll crash the process before the network request can get out.

Reply:
Removed it from here.